DataFormats/QualityControl/src/FlagReasons.cxx

```diff
@@ -0,0 +1,43 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "DataFormatsQualityControl/FlagReasons.h"
+
+#include <iostream>
+#include <tuple>
+
+ClassImp(o2::quality_control::FlagReason);
+
+namespace o2::quality_control
+{
+
+std::ostream& operator<<(std::ostream& os, FlagReason const& my)
+{
+  os << "Flag Reason: id - " << my.mId << ", name - " << my.mName << ", bad - " << (my.mBad ? "true" : "false");
+  return os;
+}
+bool FlagReason::operator==(const FlagReason& rhs) const
+{
+  return std::tie(mId, mName, mBad) == std::tie(rhs.mId, rhs.mName, rhs.mBad);
+}
+bool FlagReason::operator!=(const FlagReason& rhs) const
+{
+  return std::tie(mId, mName, mBad) != std::tie(rhs.mId, rhs.mName, rhs.mBad);
+}
+bool FlagReason::operator<(const FlagReason& rhs) const
+{
+  return std::tie(mId, mName, mBad) < std::tie(rhs.mId, rhs.mName, rhs.mBad);
```

Comment:
Why would we need a less-than and more-than?

Reply:
For some reason C++ wasn't able to deduce the other operators, even though I would expect that e.g. having `operator<` and `operator==` is enough to derive `operator>`. So I decided it is not worth time investigating for now, but I can have a look again if this bothers you.

Reply:
My question was mainly why do you want to have the "operator<" at all (and the "operator>")?

Reply:
Ah, it is to be able to sort them in TimeRangeFlagCollection with `std::set`. If we have them sorted, transforming them into Data Tags should be easier. Also, the printed output is easier to understand and with `std::set` merging is for free.

Reply:
Yes, I saw that you want to use std::set to have the merging, I did not know it needed all the operators. 
Your sorting will just be "alphabetical" on the three data members, but it won't mean much, right? Apart from being nice to be seen when printed.
Thanks for explaining.

Reply:
First we sort by start time, then end time (so chronologically) and after the other data members.

Reply:
Yes, alphabetical included the numbers - maybe one should say it differently. 
Thanks!